aws/terraform/modules/metaflow/modules/computation/locals.tf

```diff
@@ -0,0 +1,25 @@
+locals {
+  # Name of Batch service's security group used on the compute environment
+  batch_security_group_name = "${var.resource_prefix}batch-compute-environment-security-group${var.resource_suffix}"
+
+  # Prefix name of Batch CPU compute environment
+  cpu_compute_env_prefix_name = "${var.resource_prefix}cpu${var.resource_suffix}"
+
+  # Prefix name of Batch CPU compute environment
+  large_cpu_compute_env_prefix_name = "${var.resource_prefix}large-cpu${var.resource_suffix}"
+
+  # Prefix name of Batch GPU compute environment
+  gpu_compute_env_prefix_name = "${var.resource_prefix}gpu${var.resource_suffix}"
+
+  # Name of Batch Queue.
+  batch_queue_name = replace("${var.resource_prefix}${var.resource_suffix}", "--", "-")
```

Comment:
Question: Did you find something breaks without the newly added `replace(` function call?

Reply:
this is a hack. I wanted to be able to specify just a prefix/suffix and have the dashes in one place and provide logic to eliminate the dashes if the caller specified an empty prefix or suffix. This worked fine except for places where things didn't have a name, there were just prefix + suffix like the batch_queue_name. In those cases, you end up with two consecutive dashes (ie 'prefix--suffix'). I didn't want to add a name because it causes things to be recreated when you apply.

Reply:
Ahh okay I follow, makes sense with that context.

Reply:
Could: 

```suggestion
  # this ensures names that are composed of just prefix + suffix do not have duplicate dashes
  batch_queue_name = replace("${var.resource_prefix}${var.resource_suffix}", "--", "-")
```